src/nvim/api/autocmd.c

```diff
@@ -151,7 +154,13 @@ ArrayOf(DictAs(get_autocmds__ret)) nvim_get_autocmds(Dict(get_autocmds) *opts, A
     }
   }
 
-  VALIDATE((!HAS_KEY(opts, get_autocmds, pattern) || !HAS_KEY(opts, get_autocmds, buffer)),
+  Object buf = HAS_KEY(opts, get_autocmds, buf)
+               ? opts->buf
+               : (HAS_KEY(opts, get_autocmds, buffer) ? opts->buffer : NIL);
+
+  bool has_buf = HAS_KEY(opts, get_autocmds, buffer) || HAS_KEY(opts, get_autocmds, buf);
+
+  VALIDATE((!HAS_KEY(opts, get_autocmds, pattern) || !has_buf),
            "%s", "Cannot use both 'pattern' and 'buffer'", {
```

Comment:
```suggestion
           "%s", "Cannot use both 'pattern' and 'buf'", {
```

Maybe best if these errors refer to `buf` now; dunno if there's others.